### PR TITLE
aws - vpc additional related filters (subnet, nat, igw)

### PR DESCRIPTION
**Pull request by hinchliff**
Re-opening #3347.   The CCLA should be ready now?

Adding filters to the `vpc` resource, similar to the existing `VpcSecurityGroupFilter` filter:
- VpcSubnetFilter
- VpcNatGatewayFilter
- VpcInternetGatewayFilter

Also, fixed `resource_count` value_type for these RelatedResourceFilters.

Example using new filters:
```
 resource: vpc
 filters:
    - type: subnet
      value_type: resource_count
      value: 2
      op: lt
    - type: internet-gateway
      value_type: resource_count
      value: 1
      op: gte
```

### CODE REVIEW

**Comment by kapilt** [comment]
awesome! re ccla, one further request on that front, we also ask individuals doing corporate clas, to also sign the individual, as membership on corporate clas change and we use the individual for tracking. thanks!

**Comment by hinchliff** [comment]
Done!

File: c7n/filters/related.py [hunk 1]
```diff
@@ -86,16 +94,20 @@ def process_resource(self, resource, related):
...
+        if found:
+            self.add_annotations(found, resource)
 
         if op == 'or' and found:
             return True
         elif op == 'and' and len(found) == len(related_ids):
             return True
         return False
 
+    def add_annotations(self, related_ids, resource):
```
Comment (kapilt) [inline c7n/filters/related.py]: can you make this method _private (leading underscore), annotation format/schema is filter/action specific.

File: c7n/filters/related.py [hunk 2]
```diff
@@ -70,9 +70,17 @@ def process_resource(self, resource, related):
         model = self.manager.get_model()
         op = self.data.get('operator', 'or')
         found = []
+
         if self.data.get('match-resource') is True:
             self.data['value'] = self.get_resource_value(
                 self.data['key'], resource)
+
+        if self.data.get('value_type') == 'resource_count':
```
Comment (kapilt) [inline c7n/filters/related.py]: any reason why not to add this directly to value filter ?

Comment (hinchliff) [reply c7n/filters/related.py]: I'm not sure I understand what you mean.

Do you mean I should be using something from `filters/core.py` ?  There is some similar code in there, but it was not being used in `filters/related.py` ...

Comment (kapilt) [reply c7n/filters/related.py]: never mind, ic why this was done here, thanks

**Pull request merged**